pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -353,7 +354,19 @@ func (r *Reconciler) endWithNegativeCondition(ctx context.Context, app *v1beta1.
 func (r *Reconciler) patchStatus(ctx context.Context, app *v1beta1.Application, phase common.ApplicationPhase) error {
```

Comment:
I think we should write another function to mark as `patchStatusRetry`.

Not all patchStatus should retry. I'm very concern about this as it will case conflicts and inconsistent.

Reply:
The change here will only retry when conflict happens and it has a default maximal retry times as 5.
I agree that maybe not all patchStatus should retry. How about enforce all patchStatus related to workflow to retryOnConflict? /cc @leejanee @wonderflow

Reply:
Yes, workflow related patch can use this , the rest should keep as it is. We should avoid risks.